dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java

```diff
@@ -0,0 +1,315 @@
+	}
+
+	@Override
+	public WorkspaceItemRest findOne(Context context, Integer id) {
+		WorkspaceItem witem = null;
+		try {
+			witem = wis.find(context, id);
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		if (witem == null) {
+			return null;
+		}
+		return converter.fromModel(witem);
+	}
+
+	@Override
+	public Page<WorkspaceItemRest> findAll(Context context, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {
+			total = wis.countTotal(context);
+			witems = wis.findAll(context, pageable.getPageSize(), pageable.getOffset());
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<WorkspaceItemRest> page = new PageImpl<WorkspaceItem>(witems, pageable, total).map(converter);
+		return page;
+	}
+	
+	@SearchRestMethod(name = "findBySubmitter")
+	public Page<WorkspaceItemRest> findBySubmitter(@Param(value="uuid") UUID submitterID, Pageable pageable) {
+		List<WorkspaceItem> witems = null;
+		int total = 0;
+		try {		
+			Context context = obtainContext();
+			EPerson ep = epersonService.find(context, submitterID); 
+			witems = wis.findByEPerson(context, ep);
+			total = witems.size();
+		} catch (SQLException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+		Page<WorkspaceItemRest> page = new PageImpl<WorkspaceItem>(witems, pageable, total).map(converter);
+		return page;
+	}
+	
+	@Override
+	protected WorkspaceItemRest createAndReturn(Context context) {
+		WorkspaceItem source = submissionService.createWorkspaceItem(context, getRequestService().getCurrentRequest());		
+		return converter.convert(source);		
+	}
+
+	@Override
+	protected WorkspaceItemRest save(Context context, WorkspaceItemRest wsi) {
+		SubmissionConfig submissionConfig = submissionConfigReader.getSubmissionConfigByName(submissionConfigReader.getDefaultSubmissionConfigName());
+		WorkspaceItem source = converter.toModel(wsi);
+		for(int stepNum = 0; stepNum<submissionConfig.getNumberOfSteps(); stepNum++) {
+			
+			SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+			/*
+			 * First, load the step processing class (using the current
+			 * class loader)
+			 */
+			ClassLoader loader = this.getClass().getClassLoader();
+			Class stepClass;
+			try {
+				stepClass = loader.loadClass(stepConfig.getProcessingClassName());
+
+				Object stepInstance = stepClass.newInstance();
+
+				if (stepInstance instanceof AbstractProcessingStep) {
+					// load the JSPStep interface for this step
+					AbstractProcessingStep stepProcessing = (AbstractProcessingStep) stepClass
+							.newInstance();
+					stepProcessing.doProcessing(context, getRequestService().getCurrentRequest(), source);
+				} else {
+					throw new Exception("The submission step class specified by '"
+							+ stepConfig.getProcessingClassName()
+							+ "' does not extend the class org.dspace.submit.AbstractProcessingStep!"
```

Comment:
Same remark as in `WorkspaceItemConverter`

Reply:
As above.